tests/test_convergence.py

```diff
@@ -0,0 +1,101 @@
+import numpy as np
+from ase import Atoms
+import pytest
+from numpy.testing import assert_allclose
+from anisoap.representations import EllipsoidalDensityProjection
+import metatensor
+
+from scipy.special import sph_harm
+
+
+class TestGaussianConvergence:
+    sigmas = np.array([1.0, 2.0, 3.0])
+
+    @pytest.mark.parametrize("sigma", sigmas)
+    def test_single_atom_isotropic_convergence(self, sigma):
+        """
+        Test that coefficients are correct, such that the approximation using these coefficients in the expansion
+        can reasonably recreate the original atomic gaussian (in this case, an unnormalized gaussian with sigma=1)
+        """
+
+        frame = Atoms(positions=np.array([[0, 0, 0]]), numbers=[0])
+        frame.arrays["quaternions"] = np.array([[1, 0, 0, 0]])
+
+        frame.arrays[r"c_diameter[1]"] = 2 * sigma * np.ones(1)
+        frame.arrays[r"c_diameter[2]"] = 2 * sigma * np.ones(1)
+        frame.arrays[r"c_diameter[3]"] = 2 * sigma * np.ones(1)
+        frames = [frame]
+
+        # An rgw that's a little bigger than the atom gaussian sigma converges well and is sufficient for this test.
+        rgw = sigma + 0.5
+
+        max_radials = range(10)
+        errs = []
+        for max_radial in max_radials:
+            HYPER_PARAMETERS = {
+                "max_angular": 1,
+                "max_radial": max_radial,
+                "radial_basis_name": "gto",
+                "rotation_type": "quaternion",
+                "rotation_key": "quaternions",
+                "cutoff_radius": 1.0,
+                "radial_gaussian_width": rgw,
+                "basis_rcond": 1e-14,
+                "basis_tol": 1e-1,
+            }
+            representation = EllipsoidalDensityProjection(**HYPER_PARAMETERS)
+            descriptor_raw = representation.transform(frames, normalize=True)
+            descriptor = metatensor.operations.sum_over_samples(
+                descriptor_raw, sample_names="center"
+            )
+
+            def evaluate_bases(r):
```

Comment:
I do not understand what is this function for and what it's returning.

Reply:
This function outputs what the basis function $Y_{lm}*R_{nl}$ is, evaluated at the point $r$